### PR TITLE
367 change error message api

### PR DESCRIPTION
**Pull request by vdiezel**
Backend part of the changed error API.

The source of the error was that when the response from the server was a plain string and not a stringified JSON, due to the set `content-type: application/json` in the headers, axios always tried to call a JSON parse. And if this fails, it returns `null`. 

We could fix this by setting the right headers when errors occur to be `text/plain` and axios would provide the error string to the application. But I didn't like the idea of setting different content headers depending on the outcome of the request. 

We could therefore either use objects within the `createError` constructor 

`new createError(404, { message: { errorMessage: 'Device not found.' }})` 

(it has to be a nested object, because with `createError(404, { message })` the httpError-middleware would still assign the content of the message to the body, not the object `{ message: '...message' }`)

While we could do this, some prefab middleware uses plain strings in the constructor. 
So I modified our stringifyBody-middleware to react different in case of an error to create an object from the error message.

Second commit just fixes tests.

### CODE REVIEW

File: lib/middleware/stringifyResponseBody.js [hunk 1]
```diff
@@ -8,11 +8,19 @@ const middleware = (handler, next) => {
...
+  })
+
+  return next()
 }
 
 
 module.exports = () => ({
   after: (handler, next) => middleware(handler, next),
-  onError: (handler, next) => middleware(handler, next)
+  onError: (handler, next) => errorMiddleWare(handler, next)
```
Comment (vdiezel) [inline lib/middleware/stringifyResponseBody.js]: Only logic change of this PR. Rest is just fixing tests.

File: lib/middleware/stringifyResponseBody.js [hunk 2]
```diff
@@ -8,11 +8,19 @@ const middleware = (handler, next) => {
   if (!bodyIsString(handler))
     handler.response.body = JSON.stringify(handler.response.body)
 
-  next()
+  return next()
+}
+
+const errorMiddleWare = (handler, next) => {
```
Comment (DanielSeehausen) [inline lib/middleware/stringifyResponseBody.js]: middleware is one word, so camelcased: `errorMiddleware`

File: lib/middleware/stringifyResponseBody.js (hunk 1 above)
Comment (DanielSeehausen) [inline lib/middleware/stringifyResponseBody.js]: middleware is one word, so camelcased: `errorMiddleware`

Comment (DanielSeehausen) [reply lib/middleware/stringifyResponseBody.js]: (not requesting a change)

**Comment by vdiezel** [comment]
> so you did actually end up using the body , i guess it's better it's faster , however do you have in mind a client code that may use this code ? or that can be impacted by this change

Yep, here: https://github.com/ospin-web-dev/REACTor/pull/370

**Pull request merged**